crates/proof/executor/src/builder/assemble.rs

```diff
@@ -176,20 +176,16 @@ pub fn compute_receipts_root(
     // receipt encoding. In the Regolith hardfork, we must strip the deposit nonce
     // from the receipt encoding to match the receipt root calculation.
     if config.is_regolith_active(timestamp) && !config.is_canyon_active(timestamp) {
-        let receipts = receipts
-            .iter()
-            .cloned()
-            .map(|receipt| match receipt {
-                OpReceiptEnvelope::Deposit(mut deposit_receipt) => {
-                    deposit_receipt.receipt.deposit_nonce = None;
-                    OpReceiptEnvelope::Deposit(deposit_receipt)
+        ordered_trie_with_encoder(receipts, |receipt, mut buf| {
+            let to_encode: Cow<'_, OpReceiptEnvelope> = match receipt {
+                OpReceiptEnvelope::Deposit(dep) if dep.receipt.deposit_nonce.is_some() => {
+                    let mut dep2 = dep.clone();
+                    dep2.receipt.deposit_nonce = None;
+                    Cow::Owned(OpReceiptEnvelope::Deposit(dep2))
```

Comment:
I have not used `Cow` a lot, so ignore this if it is not helpful.  But isn't the benefit of using `Cow`, that can't we avoid the explicit `.clone()` and use `.to_mut()` when we need to overwrite something?

https://doc.rust-lang.org/std/borrow/enum.Cow.html#method.to_mut


```suggestion
let out cow_receipts = Cow::from(receipts).iter().map(|receipt| match receipt {
                OpReceiptEnvelope::Deposit(deposit_receipt) if deposit_receipt.deposit_nonce.is_some() => {
                    deposit_receipt.to_mut().receipt.deposit_nonce = None;;

```